### PR TITLE
support msix packages in signingscript

### PR DESCRIPTION
**Pull request by bhearsum**
Most of this patch is likely to get replaced by proper installations of msix-packaging/winsign code, once it lands upstream. The only signingscript change is updating a list of file extensions.

Not ready for review yet, just opening to keep it tracked well.

### CODE REVIEW

File: signingscript/docker.d/build_msix_packaging.sh [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
+set +e
+
+git clone https://github.com/mozilla/msix-packaging msix-packaging
+
+cd msix-packaging
+git checkout johnmcpms/signing
```
Comment (bhearsum) [inline signingscript/docker.d/build_msix_packaging.sh]: Ideally, this gets replaced by a mainlined version of this branch, in Microsoft's repository. Whether or not that happens is TBD.

Comment (bhearsum) [reply signingscript/docker.d/build_msix_packaging.sh]: This PR has been updated to point at Mozilla's repo; no movement on the Microsoft side.

**Comment by bhearsum** [comment]
@escapewindow - how do you feel about this? It's 99% picking up winsign/adding `msix-packaging` (the Mozilla fork-of-a-fork) as a dependency. The signingscript change itself is trivial.

File: signingscript/Dockerfile [hunk 2]
```diff
@@ -4,17 +4,19 @@ RUN groupadd --gid 10001 app && \
...
 RUN apt-get update \
- && apt-get install -y zipalign osslsigncode \
+ && apt-get install -y zipalign osslsigncode cmake clang \
  && apt-get clean \
  && ln -s /app/docker.d/healthcheck /bin/healthcheck
 
 COPY . /app
 RUN chown -R app:app /app
 
+RUN cd /app/signingscript/docker.d && bash build_msix_packaging.sh && cp msix-packaging/.vs/bin/makemsix /usr/bin && cp msix-packaging/.vs/lib/libmsix.so /usr/lib && cd .. && rm -rf msix-packaging
```
Comment (escapewindow) [inline signingscript/Dockerfile]: I don't love a) compile tools in the signingscript docker image, and b) multiple compiles of this tool, which may mean we get variance between docker image builds (unless we have reproducible builds here). If we're thinking this tool may stay static for a while, installing a pre-compiled tool may be preferable. If we're not sure, this is probably an acceptable short- to medium- term thing.

File: signingscript/Dockerfile [hunk 3]
```diff
@@ -4,17 +4,19 @@ RUN groupadd --gid 10001 app && \
...
+
 USER app
 WORKDIR /app
 
-RUN python -m venv /app \
+RUN cd /app && python -m venv /app \
```
Comment (escapewindow) [inline signingscript/Dockerfile]: Guessing this is because we cd'd into `/app/signingscript` above?

File: signingscript/requirements/base.in [hunk 4]
```diff
@@ -8,5 +8,5 @@ scriptworker
 # version 39.x.x
 taskcluster<40
```
Comment (escapewindow) [inline signingscript/requirements/base.in]: We can unpin this now, but doesn't have to be in this PR.

File: signingscript/src/signingscript/sign.py [hunk 5]
```diff
@@ -506,7 +506,7 @@ def _should_sign_windows(filename):
     _dont_sign = ["D3DCompiler_42.dll", "d3dx9_42.dll", "D3DCompiler_43.dll", "d3dx9_43.dll", "msvc*.dll"]
     ext = os.path.splitext(filename)[1]
     b = os.path.basename(filename)
-    if ext in (".dll", ".exe", ".msi", ".bin") and not any(fnmatch.fnmatch(b, p) for p in _dont_sign):
+    if ext in (".dll", ".exe", ".msi", ".msix", ".bin") and not any(fnmatch.fnmatch(b, p) for p in _dont_sign):
```
Comment (escapewindow) [inline signingscript/src/signingscript/sign.py:509]: I'm wondering if we should move this tuple to a constant at some point, but feeling kind of meh about it.

**Review by escapewindow — COMMENTED** [review]
> @escapewindow - how do you feel about this? It's 99% picking up winsign/adding msix-packaging (the Mozilla fork-of-a-fork) as a dependency. The signingscript change itself is trivial.

wfm, modulo review comments and the merge conflict with master.

File: signingscript/Dockerfile (hunk 2 above)
Comment (bhearsum) [reply signingscript/Dockerfile]: Where would you prefer they be built? The main alternative I can think of is someone's laptop, which seems worse. We don't have any CI stuff for our `msix-packaging` fork.

File: signingscript/Dockerfile (hunk 3 above)
Comment (bhearsum) [reply signingscript/Dockerfile]: I don't recall; I'll see if I can simplify it though.

File: signingscript/Dockerfile (hunk 2 above)
Comment (escapewindow) [reply signingscript/Dockerfile]: Hm, I suppose we could create a build-msix-packaging toolchain task that the k8s-image task downloads an artifact from. We don't have to block on this.

File: signingscript/requirements/base.in (hunk 4 above)
Comment (bhearsum) [reply signingscript/requirements/base.in]: may as well do it, it's easy :)

File: signingscript/src/signingscript/sign.py (hunk 5 above)
Comment (bhearsum) [reply signingscript/src/signingscript/sign.py:509]: yeah, i don't feel strongly either

**Review by escapewindow — APPROVED** [review]
We're punting on the msix toolchain task for now, but this would still be a good thing to do. Maybe when our branch is merged in.

**Pull request merged**